extensions/parquet/table/src/main/java/io/deephaven/parquet/table/location/ParquetColumnLocation.java

```diff
@@ -506,37 +507,28 @@ private static class LogicalTypeVisitor<ATTR extends Any>
                 final LogicalTypeAnnotation.DecimalLogicalTypeAnnotation decimalLogicalType) {
             final PrimitiveType type = columnChunkReader.getType();
             final PrimitiveType.PrimitiveTypeName typeName = type.getPrimitiveTypeName();
-            switch (typeName) {
-                case INT32:
-                    return Optional.of(ToBigDecimalFromIntPage.create(
-                            componentType, decimalLogicalType.getPrecision(), decimalLogicalType.getScale()));
-                case INT64:
-                    return Optional.of(ToBigDecimalFromLongPage.create(
-                            componentType, decimalLogicalType.getPrecision(), decimalLogicalType.getScale()));
-                case FIXED_LEN_BYTE_ARRAY:
-                case BINARY:
-                    final int encodedSizeInBytes =
-                            (typeName == PrimitiveType.PrimitiveTypeName.BINARY) ? -1 : type.getTypeLength();
-                    if (BigDecimal.class.equals(componentType)) {
-                        return Optional.of(
-                                ToObjectPage.create(
-                                        BigDecimal.class,
-                                        new BigDecimalParquetBytesCodec(
-                                                decimalLogicalType.getPrecision(), decimalLogicalType.getScale(),
-                                                encodedSizeInBytes),
-                                        columnChunkReader.getDictionarySupplier()));
-                    } else if (BigInteger.class.equals(componentType)) {
-                        return Optional.of(
-                                ToObjectPage.create(
-                                        BigInteger.class,
-                                        new BigIntegerParquetBytesCodec(encodedSizeInBytes),
-                                        columnChunkReader.getDictionarySupplier()));
-                    }
-
-                    // We won't blow up here, Maybe someone will provide us a codec instead.
-                default:
-                    return Optional.empty();
+            final int encodedSizeInBytes = (typeName == BINARY) ? -1 : type.getTypeLength();
+            if (BigDecimal.class.equals(componentType)) {
+                try {
+                    verifyScaleAndPrecision(decimalLogicalType.getScale(), decimalLogicalType.getPrecision(), typeName);
+                } catch (final IllegalArgumentException exception) {
+                    throw new TableDataException(
+                            "Invalid scale and precision for column " + name + ": " + exception.getMessage());
+                }
+                return Optional.of(ToBigDecimalPage.create(
+                        componentType,
```

Comment:
We should consider renaming `componentType` to `pageType` or `elementType` in the visitor for consistency.